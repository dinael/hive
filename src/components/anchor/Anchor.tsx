import { FC } from 'react'
import styles from './Anchor.module.scss'

export type AnchorProps = {
  text: string,
  href: string,
  target: '_self' | '_blank' | '_parent' | '_top',
  rel?: 'noopener' | 'noreferrer' | 'no-referrer' | 'noopener noreferrer',
  disabled?: boolean,
  a11yText?: string,
  safeArea?: boolean,
  className?: string,
  children?: any,
}

export const Anchor: FC<AnchorProps> = ({
  text ='',
  href ='#',
  target = '_self',
  rel = 'noreferrer',
  disabled = false,
  a11yText = '',
  safeArea = true,
  children,
  className = '',
  ...props
}: AnchorProps) => {

  const anchorStyles = `
    ${styles['anchor-component']}
    ${disabled ? styles['disabled'] : ''}
    ${safeArea ? styles['safe-area'] : ''}
    ${className || ''}
  `

  return (
    <a
      className={anchorStyles}
      href={href}
      target={target}
      rel={rel}
      aria-label={a11yText ? a11yText : undefined}
      {...props}>
      {text}{children}
    </a>
  )
}

export default Anchor
