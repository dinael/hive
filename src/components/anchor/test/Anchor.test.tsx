import { describe, test, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import { Anchor, AnchorProps } from '../Anchor'
describe('Anchor', () => {
  const defaultProps: AnchorProps = {
    text: 'Link Text',
    href: 'https://www.example.com',
    target: '_self',
    rel: 'noreferrer',
    disabled: true,
    a11yText: 'Link Description',
    safeArea: true,
    className: 'className-test',
    children: null,
  }

  test('Anchor renders snapshot', () => {
    const { container } = render(<Anchor {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  test('Anchor renders link with correct text and href', () => {
    const { container } = render(<Anchor {...defaultProps} />)
    const linkElement = container.querySelector('a')
    expect(linkElement?.getAttribute('href')).toBe('https://www.example.com')
    expect(linkElement?.textContent).toBe(defaultProps.text)
    expect(screen.getByText(defaultProps.text))
    expect(container).toMatchSnapshot()
  })

  test('Anchor renders link with target attribute', () => {
    const { container } = render(<Anchor {...defaultProps} />)
    const linkElement = container.querySelector('a')
    expect(linkElement?.getAttribute('target')).toBe(defaultProps.target)
    expect(container).toMatchSnapshot()
  })

  test('Anchor renders link with rel attribute', () => {
    const { container } = render(<Anchor {...defaultProps} />)
    const linkElement = container.querySelector('a')
    expect(linkElement?.getAttribute('rel')).toBe(defaultProps.rel)
    expect(container).toMatchSnapshot()
  })

  test('Anchor renders link with safe area class', () => {
    const { container } = render(<Anchor {...defaultProps} />)
    const linkElement = container.querySelector('a')
    expect(linkElement?.className).toContain('safe-area')
    expect(container).toMatchSnapshot()
  })

  test('Anchor renders link with disabled class', () => {
    const { container } = render(<Anchor {...defaultProps} />)
    const linkElement = container.querySelector('a')
    expect(linkElement?.className).toContain('disabled')
    expect(container).toMatchSnapshot()
  })

  test('Anchor renders link with custom class', () => {
    const { container } = render(<Anchor {...defaultProps} />)
    const linkElement = container.querySelector('a')
    expect(linkElement?.className).toContain(defaultProps.className)
    expect(container).toMatchSnapshot()
  })

  test('Anchor renders link with children', () => {
    const { container } = render(<Anchor {...defaultProps}>Child Element</Anchor>)
    const linkElement = container.querySelector('a')
    expect(linkElement?.textContent).toBe('Link TextChild Element')
    expect(screen.getByText('Link TextChild Element'))
    expect(container).toMatchSnapshot()
  })

  test('Anchor renders link with a11yText', () => {
    const { container } = render(<Anchor {...defaultProps}  />)
    const linkElement = container.querySelector('a')
    expect(linkElement?.getAttribute('aria-label')).toBe(defaultProps.a11yText)
    expect(container).toMatchSnapshot()
  })
})
