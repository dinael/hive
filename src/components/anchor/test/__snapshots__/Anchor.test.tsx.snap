// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Anchor > Anchor renders link with a11yText 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > Anchor renders link with children 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
    Child Element
  </a>
</div>
`;

exports[`Anchor > Anchor renders link with correct text and href 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > Anchor renders link with custom class 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > Anchor renders link with disabled class 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > Anchor renders link with rel attribute 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > Anchor renders link with safe area class 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > Anchor renders link with target attribute 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > Anchor renders snapshot 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > renders link with a11yText 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > renders link with children 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
    Child Element
  </a>
</div>
`;

exports[`Anchor > renders link with correct text and href 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > renders link with custom class 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > renders link with disabled class 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > renders link with rel attribute 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > renders link with safe area class 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > renders link with target attribute 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;

exports[`Anchor > renders snapshot 1`] = `
<div>
  <a
    aria-label="Link Description"
    class="
        anchor-component
        disabled
        safe-area
        disabled
        className-test
      "
    href="https://www.example.com"
    rel="noreferrer"
    target="_self"
  >
    Link Text
  </a>
</div>
`;
